src/server/lobby.rs

```diff
@@ -81,6 +82,14 @@ fn propagate_lobby_status(
             room_id,
             &room_manager,
         );
+
+        // Send the seed message
+        let seed: u32 = rand::thread_rng().gen();
+        let _ = connection_manager.send_message_to_room::<ReliableChannel, _>(
+            &SeedMessage(seed),
+            room_id,
+            &room_manager,
+        );
```

Comment:
This is not ideal because each client will receive a newly generated "random" seed.

My suggestion will be to generate a new seed only when a new lobby is created. Then store the seed inside the lobby (can be a component attached to the lobby entity or a new field inside the `Lobby` component itself).

Also, we only need to send the seed to the newly joined client instead of the entire room. Use `send_message` ✅ instead of `send_message_to_room` ❌ .